runtime/lua/vim/lsp.lua

```diff
@@ -1365,16 +1365,26 @@ end
 
 --- Provides an interface between the built-in client and a `foldexpr` function.
 ---
---- To use, check for the "textDocument/foldingRange" capability in an
---- |LspAttach| autocommand. Example:
+--- To use, set 'foldmethod' to "expr" and set the value of 'foldexpr':
 ---
 --- ```lua
+--- vim.o.foldmethod = 'expr'
+--- vim.o.foldexpr = 'v:lua.vim.lsp.foldexpr()'
+--- ```
+---
+--- Or use it on demand by checking for the "textDocument/foldingRange"
+--- capability in an |LspAttach| autocommand. Example:
+---
+--- ```lua
+--- -- General folding preferences
+--- vim.o.foldmethod = 'expr'
+--- vim.o.foldexpr = 'v:lua.vim.treesitter.foldexpr()'
+--- -- Enable LSP folding if a client supports it.
```

Comment:
```suggestion
--- vim.o.foldmethod = 'expr'
```

Reply:
This mixes up treesitter and LSP; I think we should focus on the latter here.

Otherwise do
```lua
vim.o.foldmethod = 'expr'
-- default to treesitter folding
vim.o.foldexpr = 'v:lua.vim.treesitter.foldexpr()'
-- prefer LSP folding if client supports it
```